src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -16,19 +19,38 @@ internal enum WebHeaderCollectionType : byte
         WebResponse,
     }
 
-    public sealed class WebHeaderCollection : IEnumerable
+    [Serializable]
+    public class WebHeaderCollection : NameValueCollection, ISerializable
     {
         private const int ApproxAveHeaderLineSize = 30;
         private const int ApproxHighAvgNumHeaders = 16;
 
-        // Lazily initialized fields.
-        private List<string> _entriesList;
-        private Dictionary<string, string> _entriesDictionary;
-        private string[] _allKeys;
+        // Lazily initialized field.
+        private NameValueCollection _innerCollection;
```

Comment:
Nit: please group the fields, rather than putting the fields next to the members that access them